### PR TITLE
Add bearer token provider

### PR DESCRIPTION
**Pull request by lmazuel**
Add a get_bearer_token_provider to azure-identity

### CODE REVIEW

**Comment by joshfree** [comment]
Flagging as `Discuss in Office Hours` for Wednesday this week

**Review by xiangyan99 — APPROVED** [review]
Can you also update the changelog?

File: sdk/identity/azure-identity/azure/identity/_bearer_token_provider.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    """Returns a callable that provides a bearer token.
+
+    It can be used for instance to write code like:
+
+    .. code-block:: python
+
+        from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+
+        credential = DefaultAzureCredential()
+        bearer_token_provider = get_bearer_token_provider(credential, "https://storage.azure.com/.default")
```
[the remark this answers is not in the document]
Comment (lmazuel) [reply sdk/identity/azure-identity/azure/identity/_bearer_token_provider.py:27]: Good point, let's showcase our primary scenario indeed

File: sdk/identity/azure-identity/azure/identity/aio/_bearer_token_provider.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        credential = DefaultAzureCredential()
+        bearer_token_provider = get_bearer_token_provider(credential, "https://storage.azure.com/.default")
+
+
+        # Usage
+        request.headers["Authorization"] = "Bearer " + await bearer_token_provider()
+
+    :param credential: The credential used to authenticate the request.
+    :type credential: ~azure.core.credentials.TokenCredential
```
[the remark this answers is not in the document]
Comment (lmazuel) [reply sdk/identity/azure-identity/azure/identity/aio/_bearer_token_provider.py:34]: Fixed

**Pull request merged**